TheMovieDb/Search/Controller/SearchViewController.swift

```diff
@@ -7,23 +7,108 @@
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        configureUI()
-        // Do any additional setup after loading the view.
+        configureSearchController()
+        configureTableView()
+       
     }
     
-    private func configureUI() {
-        view.backgroundColor = .systemBackground
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        searchController.isActive = true
     }
     
     // MARK: - Helpers
-        
-    // MARK: - Actions
+    
+    func configureTableView() {
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: reuseIdentifier)
+        tableView.rowHeight = 64
+    }
+    
+    func configureSearchController() {
+        searchController.searchBar.delegate = self
+        searchController.delegate = self
+        searchController.obscuresBackgroundDuringPresentation = false
+        searchController.hidesNavigationBarDuringPresentation = false
+        searchController.searchBar.placeholder = "Search"
+        navigationItem.searchController = searchController
+        navigationItem.titleView?.isHidden = true
+        definesPresentationContext = false
+
+    }
+    
+    // MARK: - API
+    
+}
 
+// MARK: - UITableViewDataSource
+extension SearchViewController {
+   
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return moviesSearch.count
+    }
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier, for: indexPath)
+        cell.textLabel?.text = moviesSearch[indexPath.row].title
+        let url = URL(string: MovieConst.imageCDN + (moviesSearch[indexPath.row].posterPath ?? "") )
+        cell.imageView?.kf.setImage(with: url)
+        cell.imageView?.contentMode = .scaleAspectFit
+        return cell
+    }
+}
+
+// MARK: - UITableDelegate
+extension SearchViewController {
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let movie =  moviesSearch[indexPath.row]
+        let controller = DetailViewController(with: movie)
+        navigationController?.pushViewController(controller, animated: true)
+    }
+}
+
+extension SearchViewController: UISearchControllerDelegate {
+    func didPresentSearchController(_ searchController: UISearchController) {
+        DispatchQueue.main.async {
+            searchController.searchBar.becomeFirstResponder()
+        }
+    }
+}
+
+// MARK: - UISearchBarDelegate
+extension SearchViewController: UISearchBarDelegate {
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let text = searchBar.text else {return}
+        
+        let parameter = APIParameters(query: text)
+        MovieAPI.shared.fetchData(endPoint: .search, with: parameter, completion: {(response: Result<Movies, Error>) in
```

Comment:
Singletons shouldn't be called directly instead create a var on the controller, with a protocol to allow to dependency injection to be testable, this will also reduce the amount of work when you move into another architecture

Reply:
I've changed the code with a protocol, now  it passing with  dependency injection 😁
```suggestion
        let parameter = APIParameters(query: text)
        self.moviesSearch = searchData.getMovie(with: parameter)
        DispatchQueue.main.async {
            self.tableView.reloadData()
        }
```